### PR TITLE
chore: Updates nightly tests to TF 1.9.x

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Updates nightly tests to TF 1.9.x

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .github/workflows/test-suite.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ on:
     inputs:
       terraform_matrix:
         description: 'Terraform version matrix (JSON array)'
-        default: '["1.8.x", "1.7.x", "1.6.x", "1.5.x", "1.4.x", "1.3.x", "1.2.x"]'
```
Comment (EspenAlbert) [inline .github/workflows/test-suite.yml:9]: Is this change intentional? Removing 1.5.x down to 1.2.x ?

File: .github/workflows/test-suite.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ on:
...
+        default: '["1.9.x", "1.8.x", "1.7.x", "1.6.x"]'
```
Comment (maastha) [inline .github/workflows/test-suite.yml]: We're still supporting versions as low as 1.3.x as per https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2370/files until 9/30, could you confirm why these are being removed?

File: .github/workflows/update_tf_compatibility_matrix.yml [hunk 3]
```diff
@@ -30,4 +30,4 @@ jobs:
           commit-message: "doc: Updates Terraform Compatibility Matrix documentation"
           delete-branch: true
           branch: terraform-compatibility-matrix-update
-          body: "Automatic updates for Terraform Compatibility Matrix documentation. **Action Required**: Update .tools-version file and TF_VERSION_LATEST GitHub environment variable if needed."
+          body: "Automatic updates for Terraform Compatibility Matrix documentation. **Action Required**: Update test-suite.yml and .tools-version files, and TF_VERSION_LATEST GitHub environment variable if needed."
```
Comment (maastha) [inline .github/workflows/update_tf_compatibility_matrix.yml]:
```suggestion
          body: "Automatic updates for Terraform Compatibility Matrix documentation. **Action Required**: Update test-suite.yml, .tools-version files, and TF_VERSION_LATEST GitHub environment variable if needed."
```

File: .github/workflows/test-suite.yml (hunk 2 above)
Comment (lantoli) [reply .github/workflows/test-suite.yml]: my bad, reverted, thx

File: .github/workflows/test-suite.yml (hunk 1 above)
Comment (lantoli) [reply .github/workflows/test-suite.yml:9]: my bad, reverted, thx

**Review by EspenAlbert — APPROVED** [review]
Thanks for the quick changes. LGTM 👍
